### ISSUE TITLE
RPC server still has handlers registered in dtests involving repair-based operations

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#7262]
Seen consistently in the following dtests since:
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/611/testReport/bootstrap_test/TestBootstrap/killed_wiped_node_cannot_join_test_2/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/611/testReport/bootstrap_test/TestBootstrap/shutdown_wiped_node_cannot_join_test/
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/611/testReport/replace_address_test/TestReplaceAddress/replace_nonexistent_node_test_2/

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/611/artifact/logs-release.2/1600544865570_bootstrap_test.TestBootstrap.killed_wiped_node_cannot_join_test/node4.log
```
INFO  2020-09-19 19:37:36,429 [shard 0] init - Shutting down messaging service
ERROR 2020-09-19 19:37:36,429 [shard 0] messaging_service - RPC server still has handlers registered
ERROR 2020-09-19 19:37:36,429 [shard 0] messaging_service -  - 6
ERROR 2020-09-19 19:37:36,429 [shard 0] messaging_service -  - 7
ERROR 2020-09-19 19:37:36,429 [shard 0] messaging_service -  - 8
ERROR 2020-09-19 19:37:36,429 [shard 0] messaging_service -  - 9
ERROR 2020-09-19 19:37:36,429 [shard 0] messaging_service -  - 10
ERROR 2020-09-19 19:37:36,429 [shard 0] messaging_service -  - 44
Aborting on shard 0.
Backtrace:
  0x0000000002a72f42
  0x0000000002a735d0
  0x0000000002a73875
  0x0000000002a738c0
  0x00007f9adf393a8f
  /jenkins/workspace/scylla-master/dtest-release/scylla/.ccm/scylla-repository/311b6b827c87af988f51ad5f86afb68f0cd01a18/scylla//opt/scylladb/libreloc/libc.so.6+0x000000000003c9e4
  /jenkins/workspace/scylla-master/dtest-release/scylla/.ccm/scylla-repository/311b6b827c87af988f51ad5f86afb68f0cd01a18/scylla//opt/scylladb/libreloc/libc.so.6+0x0000000000025894
  0x0000000001e06528
  0x0000000001e5a50a
  0x0000000001e06262
  0x00000000019ab7a8
  0x0000000002cf44d6
  0x00000000019dad02
  0x0000000001ebd9ab
  0x0000000000c7363a
  0x0000000000cc97b2
  0x0000000000cdbba3
  0x0000000002d0dc7c
```

I'm not sure why there isn't a core dump

**Comment by xemul** [comment]
Cute. Gossiper registers verbs handlers in two places -- `start_gossiping` and `do_shadow_round`. Unregisters in on -- `stop_gossiping`. The latter is only called on (aborted)? shutdown if `_enabled` is set to true inside `start_gossiping`.

**Comment by bhalevy** [comment]
Still happens, reopening...

https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/613/artifact/logs-release.2/1600801102911_replace_address_test.TestReplaceAddress.replace_nonexistent_node_test/node4.log

**Comment by xemul** [comment]
@bhalevy , the fix is a75b0486, but git log says that it was committed _after_ 75e72d18

**Comment by bhalevy** [comment]
Sorry, I got confused by the (incomplete) original fix: e43d6d146013a3cd0b75cfb070254bc8da9c0f5d

**Comment by avikivity** [comment]
Minor, not backporting.
